common/persistence/dataInterfaces.go

```diff
@@ -891,6 +891,8 @@ type (
 	ReadHistoryBranchByBatchResponse struct {
 		// History events by batch
 		History []*historypb.History
+		// TransactionID for relevant History batch
+		TransactionIDs []int64
```

Comment:
how about changing 
```go
History []*historypb.History
TransactionIDs []int64
```

to something like
```go
History []History
```
where
```go
History struct {
  TransactionID int64
  PrevTransactionID int64
  Events *historypb.History
}

Reply:
basically do not use 2 slices with equal length, but one slice of structs containing all data to be used

Reply:
I would suggest to make it a separate PR. It blows up into too big of a change.